### ISSUE TITLE
MQTT Connection refused - not authorized

### ISSUE THREAD

**Issue by BenoitDefraigne** [joleys/niko-home-control-II#31]
Hello,

I installed the nhc2 integration but can't get it working (I don't even get the installation wizard). When looking at the HA logs I find the error generated by coco.py line126 "Exception: Connection refused - not authorised", systematically.

What can be wrong? 

Logger: root
Source: custom_components/nhc2/nhccoco/coco.py:126
Integration: Niko Home Control II (documentation, issues)
First occurred: 7:14:47 PM (1 occurrences)
Last logged: 7:14:47 PM

Uncaught thread exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.10/site-packages/paho/mqtt/client.py", line 3591, in _thread_main
    self.loop_forever(retry_first_connection=True)
  File "/usr/local/lib/python3.10/site-packages/paho/mqtt/client.py", line 1756, in loop_forever
    rc = self._loop(timeout)
  File "/usr/local/lib/python3.10/site-packages/paho/mqtt/client.py", line 1164, in _loop
    rc = self.loop_read()
  File "/usr/local/lib/python3.10/site-packages/paho/mqtt/client.py", line 1556, in loop_read
    rc = self._packet_read()
  File "/usr/local/lib/python3.10/site-packages/paho/mqtt/client.py", line 2439, in _packet_read
    rc = self._packet_handle()
  File "/usr/local/lib/python3.10/site-packages/paho/mqtt/client.py", line 3039, in _packet_handle
    return self._handle_connack()
  File "/usr/local/lib/python3.10/site-packages/paho/mqtt/client.py", line 3138, in _handle_connack
    on_connect(
  File "/config/custom_components/nhc2/nhccoco/coco.py", line 126, in _on_connect
    raise Exception(MQTT_RC_CODES[rc])
**Exception: Connection refused - not authorised**

Home Assistant 2022.9.7
Supervisor 2022.09.1
Operating System 9.0
Frontend 20220907.2 - latest

**Comment by MirandaPoth** [comment]
‘Connection refused’ usually means you have the incorrect user name or password.

**Comment by BenoitDefraigne** [comment]
Thank you Miranda.
I am puzzled because the username and password in configuration.yaml seem correct, double and triplechecked ;) _Are we indeed  talking about the credentials on the NHCII hub_  ? 
In my case, i use the  touchscreen profile defined in the NHC2 programming software:

<img width="173" alt="image" src="https://user-images.githubusercontent.com/109746853/193460196-c3a08c61-4736-4167-990b-b14a852c2426.png">

I tried all combinations of quoted and not quoted parameters, nothing helps.

Also, I doubt if it is related to credentials beacause when looking at the code I found in nhccoco/login_validation.py (that seem to test the login on mqtt (NHC2?) ) the list of rc values, and there is a dedicated return code (4) for incorrect credentials. Code (5) is the one I get, looking at the Exception text.

Try to connect with given parameters
        The return indicates success or not:
            0: Connection successful
            1: Connection refused - incorrect protocol version
            2: Connection refused - invalid client identifier
            3: Connection refused - server unavailable
            4: Connection refused - bad username or password
            5: Connection refused - not authorised
            6-255: Currently unused.

So, still unresolved... any idea to help, anyone?

**Comment by MirandaPoth** [comment]
The username should be 'hobby', not 'default'. And I assume you've got the long password from the Niko API site?

If that doesn't help, here's a long shot that it worked for me once, and it took me ages to find it.

I was convinced I had the correct username and password, and I felt that HA was caching it somewhere. Eventually I looked in a file in the .storage folder called core.config_entries, and found that it had a PREVIOUS password in there. I copied & pasted in the correct password, restarted HA, and it worked ...
